File: StoringData/app/src/main/java/com/example/storingdata/MainActivity.java
package com.example.storingdata;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    EditText editText;
    TextView textView;
    SharedPreferences sharedPreferences;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        editText = findViewById(R.id.editText);
        textView = findViewById(R.id.textView1);

        sharedPreferences = this.getSharedPreferences("com.example.storingdata", Context.MODE_PRIVATE);
        int storedAge = sharedPreferences.getInt("StoredAge",0);
        if(storedAge == 0){
            textView.setText("Your Age: ");
        }else{
            textView.setText("Your Age: " + storedAge);
        }
    }
    public void save(View view){
        if(!editText.getText().toString().equals("")){
            int userAge = Integer.parseInt(editText.getText().toString());
            textView.setText("Your Age: " + userAge);
            sharedPreferences.edit().putInt("StoredAge",userAge).apply();
        }
    }
    public void delete(View view){
        int storedData = sharedPreferences.getInt("StoredAge",0);
        if(storedData != 0){
            sharedPreferences.edit().remove("StoredAge").apply();
            textView.setText("Your Age: ");
        }
    }

}

File: StoringData2/app/src/main/java/com/muratcakir/storingdata/MainActivity.kt
package com.muratcakir.storingdata

import android.content.Context
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    lateinit var sharedPreferences: SharedPreferences
    var ageFromPreferences : Int? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        sharedPreferences = this.getSharedPreferences("com.muratcakir.storingdata", Context.MODE_PRIVATE)

        ageFromPreferences = sharedPreferences.getInt("age",-1)

        if(ageFromPreferences == -1){
            textView.text = "Your Age: "
        }else{
            textView.text = "Your Age: " + ageFromPreferences
        }
    }
    fun Save (view: View){

        val myAge = editText.text.toString().toIntOrNull()

        if(myAge != null){
            textView.text = "Your Age: " + myAge
            sharedPreferences.edit().putInt("age",myAge).apply()
        }

    }
    fun Delete (view: View){

        ageFromPreferences = sharedPreferences.getInt("age",-1)

        if(ageFromPreferences != -1){
            sharedPreferences.edit().remove("age").apply()
            textView.text = "Your Age: "
        }

    }
}
